tools/qac_compile_commands.py

```diff
@@ -0,0 +1,28 @@
+import sys, subprocess, json, os
```

Comment:
Let's add a `__main__` function that intializes `refresh_compile_commands` and `workspace_dir`, and then passes those parameters to a `main()` function to tidy up this script.

Also make sure it's formatted with `black`.

Lastly add some doc on what this script in doing (similiar to the doc I added in the qac script).